File: atr_pkg/src/motor_node.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

rclcpp::Node::SharedPtr node;

void ultra_sound_callback(const std_msgs::msg::String msg)
{
    RCLCPP_INFO(node->get_logger(),
                "I am the motor node and received a message: %s",
                msg.data.c_str());
}

void aml_callback(const std_msgs::msg::String msg)
{
    RCLCPP_WARN(node->get_logger(),
                "I am the motor node and received a message: %s",
                msg.data.c_str());
}


int main(int argc, char const *argv[])
{
    rclcpp::init(argc, argv);

    printf("Hello World, I am the motor node!\n");

    node = rclcpp::Node::make_shared("motor_node");

    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_ultra_sound = node->create_subscription<std_msgs::msg::String>("ultra_sound_topic", 10, ultra_sound_callback);
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr sub_aml = node->create_subscription<std_msgs::msg::String>("aml_topic", 10, aml_callback);


    rclcpp::spin(node);

    return 0;
}

File: atr_pkg/atr_pkg/motor_node.py
#!/usr/bin/env python3
#conding: utf-8


import rclpy
from std_msgs.msg import String

node = None

def ultra_sound_callback(msg):
    global node
    node.get_logger().info("I am the motor node and received a message: " + msg.data)

def aml_callback(msg):
    global node
    node.get_logger().warn("I am the motor node and received a message: " + msg.data)


def main(args=None):
    global node
    rclpy.init(args=None)
    node = rclpy.create_node('motor_node')

    print("Hello World, I am the motor node!\n")

    sub_ultra_sound = node.create_subscription(String, 'ultra_sound_topic', ultra_sound_callback, 10)
    sub_aml = node.create_subscription(String, 'aml_topic', aml_callback, 10)

    rclpy.spin(node)

    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

